streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/RackAwareTaskAssignor.java

```diff
@@ -486,12 +486,12 @@ public long optimizeStandbyTasks(final SortedMap<UUID, ClientState> clientStates
                         .sorted()
                         .collect(Collectors.toList());
 
-                    final Map<TaskId, UUID> taskClientMap = new HashMap<>();
                     final List<UUID> clients = Stream.of(clientList.get(i), clientList.get(j))
                         .sorted().collect(
                             Collectors.toList());
-                    final Map<UUID, Integer> originalAssignedTaskNumber = new HashMap<>();
 
+                    final Map<TaskId, UUID> taskClientMap = new HashMap<>();
+                    final Map<UUID, Integer> originalAssignedTaskNumber = new HashMap<>();
```

Comment:
I moved this lower down to show them clearly initialized empty right before the function call. It turns out there parameters are used as outputs, which is why I added the `AssignmentGraph` inner class to the `TaskAssignmentUtils`, in order to simplify the usage of the graph functions.